src/dune_rules/merlin.ml

```diff
@@ -145,6 +156,11 @@ module Processed = struct
           print "# FLG";
           List.iter flags ~f:(fun f -> printf " %s" (quote_for_dot_merlin f));
           print "\n");
+    let () =
+      match mode with
+      | `Ocaml -> ()
+      | `Melange -> print "# FLG -ppx melc -as-ppx -bs-jsx 3\n"
```

Comment:
I think we can use the same machinery above, with `quote_for_dot_merlin`. See my comment in the test.

Reply:
Please check https://github.com/ocaml/dune/pull/6476/commits/5c34973f817c0dab8da113fcb6849b3898eb0520.